mlflow/pyfunc/__init__.py

```diff
@@ -766,7 +766,9 @@ def _predict(self, data: PyFuncInput, params: Optional[Dict[str, Any]] = None) -
             Model predictions as one of pandas.DataFrame, pandas.Series, numpy.ndarray or list.
         """
         data, params = self._validate_prediction_input(data, params)
-        if inspect.signature(self._predict_fn).parameters.get("params"):
+        if (
+            v := inspect.signature(self._predict_fn).parameters.get("params")
+        ) and v.kind != inspect.Parameter.VAR_KEYWORD:
```

Comment:
Shouldn't we apply this change here as well?

https://github.com/mlflow/mlflow/blob/b96b36ba5be14f6b70955f72527b38aaea8cac71/mlflow/pyfunc/model.py#L652

Reply:
```suggestion
        params_arg = inspect.signature(self._predict_fn).parameters.get("params")
        if params_arg and params_arg.kind != inspect.Parameter.VAR_KEYWORD:
```

for better readability